scripts/joinmarket-qt.py

```diff
@@ -1462,10 +1462,11 @@ def create_menu(self, position):
             txt = str(item.text(0))
             if validate_address(txt)[0]:
                 address_valid = True
-            if "EXTERNAL" in txt:
-                parsed = txt.split()
-                if len(parsed) > 1:
-                    xpub = parsed[1]
+
+            parsed = txt.split()
+            if len(parsed) > 1:
+                if parsed[-1].startswith("xpub") or parsed[-1].startswith("ypub") or parsed[-1].startswith("zpub"):
+                    xpub = parsed[-1]
```

Comment:
Both the pre-existing code (relying on "EXTERNAL" and list position of the xpub) and your new code (relying on having a list of items more than 1 and that the final one is where the xpub is stored, and also the repeated check for [x,y,z]pub which I think could be abstracted) are a bit brittle in how they assume the structure of the tree items we're creating.

I don't think there's anything actually *wrong* though. Also a lot of the code in this script is like that, anyway.

Reply:
(You could just get rid of repeated `startswith` with `if any([parsed[-1].startswith[prefix] for prefix in ["xpub",...]])` as one simplifier, and/or push out the list of *pub strings somewhere else)

Reply:
> I don't think there's anything actually wrong though. Also a lot of the code in this script is like that, anyway.

Going to eventually update the `is_extended_public_key` method to perform the base58 decoding and compare the check sum. Probably in a following up PR after this one.